dhis2/data-visualizer-app
# chore: configure continuous delivery workflows

There are 4 workflows:

- verify PR (dhis2-verify-app.yml)
-- build, lint, test, e2e-prod

might remove this one
- verify commit to dev (dhis2-verify-app.yml)
-- build, lint, test, e2e-prod, report-failure-to-slack

- verify commit to master (dhis2-verify-app.yml)
-- build, lint, test, e2e-prod, release, report-failure-to-slack

- nightly (nightly.yml)
-- e2e-dev, report-failure

In addition:
* e2e-prod and e2e-dev are reusable workflows and are called from dhis2-verify-app and nightly.
* removed `uses: c-hive/gha-yarn-cache@v1` since it is deprecated and setup-node handles that work.
* updated action versions and node versions
* cypress test version tagging for features and bugs has been added (copied from line-list)
